stored_requests/fetcher.go

```diff
@@ -210,18 +211,27 @@ func (f *fetcherWithCache) FetchResponses(ctx context.Context, ids []string) (da
 	return
 }
 
-func (f *fetcherWithCache) FetchAccount(ctx context.Context, accountID string) (account json.RawMessage, errs []error) {
+func (f *fetcherWithCache) FetchAccount(ctx context.Context, accountDefaultJSON json.RawMessage, accountID string) (account *config.Account, errs []error) {
 	accountData := f.cache.Accounts.Get(ctx, []string{accountID})
 	// TODO: add metrics
-	if account, ok := accountData[accountID]; ok {
+	if accountJSON, ok := accountData[accountID]; ok {
 		f.metricsEngine.RecordAccountCacheResult(metrics.CacheHit, 1)
+		account = &config.Account{}
+		err := json.Unmarshal(accountJSON, account)
+		if err != nil {
+			return nil, []error{err}
+		}
 		return account, errs
 	} else {
 		f.metricsEngine.RecordAccountCacheResult(metrics.CacheMiss, 1)
 	}
-	account, errs = f.fetcher.FetchAccount(ctx, accountID)
+	account, errs = f.fetcher.FetchAccount(ctx, accountDefaultJSON, accountID)
 	if len(errs) == 0 {
-		f.cache.Accounts.Save(ctx, map[string]json.RawMessage{accountID: account})
+		accountJSON, err := json.Marshal(account)
```

Comment:
Once you make those above changes, you won't have to do JSON marshalling here before saving it in the cache as you'll already have it as raw JSON string from `FetchAccount`

Reply:
Reverted the account marshalling changes.